qiskit_dynamics/pulse/pulse_simulator.py

```diff
@@ -0,0 +1,330 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+import time
+import datetime
+import uuid
+
+from typing import Dict, Iterable, List, Optional, Union
+import numpy as np
+
+from qiskit import pulse
+from qiskit.pulse import Schedule, ScheduleBlock
+from qiskit.pulse.channels import AcquireChannel, DriveChannel, MeasureChannel, ControlChannel
+from qiskit.pulse.transforms.canonicalization import block_to_schedule
+from qiskit.providers.backend import BackendV2
+from qiskit.result import Result
+
+from qiskit import QiskitError
+from qiskit.quantum_info import Statevector, DensityMatrix
+
+from qiskit_dynamics import Solver
+from qiskit_dynamics.array import Array
+from qiskit_dynamics.models import HamiltonianModel
+
+from .dynamics_job import DynamicsJob
+from .pulse_utils import _get_dressed_state_decomposition, _get_lab_frame_static_hamiltonian
+
+
+class PulseSimulator(BackendV2):
+    def __init__(
+        self,
+        solver: Solver,
+        subsystem_dims,
+        subsystem_labels: Optional[List[int]] = None,
+        name: Optional[str] = 'PulseSimulator',
```

Comment:
I dont think name is necessary here, unless its a base class. You want init to add `target=None` so you can pass in a custom target, and `**options` for passing arbitrary things to options (and solver, labels etc can just be handled by options).

For handling options I would do something like AerBackend does with configuration/properties of V1, where it can allow configuration of anything in target through `set_options` (unless you explicitly want to forbid it), or its own internal options.